contract/views/contract.xml

```diff
@@ -85,6 +87,9 @@
             <field name="model">account.analytic.account</field>
             <field name="inherit_id" ref="analytic.view_account_analytic_account_search"/>
             <field name="arch" type="xml">
+                <field name="partner_id" position="attributes">
+                    <attribute name="operator">child_of</attribute>
```

Comment:
Better to create another different line with this behavior, so that you can search by exact partner, or by the partner and its childs.